src/core/server/i18n/get_kibana_translation_files.ts

```diff
@@ -17,26 +17,27 @@
  * under the License.
  */
 
-import { KibanaConfig } from '../kbn_server';
-import { fromRoot } from '../../../core/server/utils';
-import { I18N_RC } from './constants';
+import { basename } from 'path';
+import { fromRoot } from '../utils';
 import { getTranslationPaths } from './get_translation_paths';
 
-export async function getKibanaTranslationPaths(config: Pick<KibanaConfig, 'get'>) {
-  return await Promise.all([
+export const getKibanaTranslationFiles = async (
+  locale: string,
+  pluginPaths: string[]
+): Promise<string[]> => {
+  const translationPaths = await Promise.all([
     getTranslationPaths({
       cwd: fromRoot('.'),
-      glob: `*/${I18N_RC}`,
+      nested: true,
     }),
-    ...(config.get('plugins.paths') as string[]).map((cwd) =>
-      getTranslationPaths({ cwd, glob: I18N_RC })
-    ),
-    ...(config.get('plugins.scanDirs') as string[]).map((cwd) =>
-      getTranslationPaths({ cwd, glob: `*/${I18N_RC}` })
-    ),
+    ...pluginPaths.map((pluginPath) => getTranslationPaths({ cwd: pluginPath, nested: false })),
```

Comment:
There is a slight change of behavior here:

Legacy was just reading the plugin config and scanning the `scanDirs` (1 lvl deep) and `paths` (root lvl) for i18n rc files.

I chose to change that by retrieving the list of enabled plugins from the plugin service, and check if there is a i18n file at the root level of each plugin instead. This is basically the same, except that we are not loading translation files for disabled plugins (note that this only concerns 3rd parties plugin, as kibana translations are all defined from a single `x-pack/.i18nrc.json` file anyway). Overall, I think it should not really impact 3rd parties, and not loading translations for disabled plugins does make sense imho. (This is also way better at the code level in term of responsibilities per service. I was really awkward to load the `plugins` config from the `i18n` service)